From 88ead890c0de9d6fe2304c4a17d6baa8eb2b4510 Mon Sep 17 00:00:00 2001
From: Fancy Fang <chen.fang@nxp.com>
Date: Thu, 30 Aug 2018 15:13:57 +0800
Subject: [PATCH 4616/5242] MLK-19400-1 drm/bridge: sec-dsim: correct one fail
 branching

commit  6a31777e043a1b34a2248bd7c4fb6167e1f673b0 from
https://source.codeaurora.org/external/imx/linux-imx.git

Obviously, DRM panel prepare is done after the DSIM PLL config,
so when PLL config failed, the 'sec_mipi_dsim_bridge_enable()'
should return directly instead of goto DRM panel unprepare.

Signed-off-by: Fancy Fang <chen.fang@nxp.com>
(cherry picked from commit 9c77865d6ec98ff60a82f03743fd797d082634cc)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/drm/bridge/sec-dsim.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/bridge/sec-dsim.c b/drivers/gpu/drm/bridge/sec-dsim.c
index 5a1f8ca..ea4966a 100644
--- a/drivers/gpu/drm/bridge/sec-dsim.c
+++ b/drivers/gpu/drm/bridge/sec-dsim.c
@@ -1187,7 +1187,7 @@ static void sec_mipi_dsim_bridge_enable(struct drm_bridge *bridge)
 	ret = sec_mipi_dsim_config_pll(dsim);
 	if (ret) {
 		dev_err(dsim->dev, "dsim pll config failed: %d\n", ret);
-		goto panel_unprepare;
+		return;
 	}
 
 	/* config dphy timings */
-- 
1.7.9.5

